src/Network/Broadcast/OutboundQueue.hs

```diff
@@ -642,7 +655,18 @@ registerQueueMetrics OutQ{..} store = do
                                 store
 
     metric :: [String] -> Text
-    metric = T.pack . intercalate "." . (["queue", qSelf] ++)
+    metric =
+        let initialList = ["queue", qSelf]
+            -- If we have a namespace, add it to the beginning of the
+            -- initial list, so that the final metric name will be
+            -- correctly namespaced.
+            -- E.g.
+            -- let namespaceMb = Just "mynamespace"
+            -- let qSelf       = "test"
+            -- metric ["InFlight"]
+            -- >>> mynamespace.queue.test.InFlight
+            tokens      = maybe initialList (: initialList) namespaceMb
```

Comment:
simpler:

     let prefix = maybeToList namespaceMb ++ ["queue", qSelf]
      in T.pack . intercalate "." . (prefix ++)

Reply:
believe it or not TIL about `maybeToList`, thanks! 😀  (I get constantly amazed by the sheer number of useful combinator in `base`).